PBM-887 Documented pbm diagnostic command (#226)
percona/pbm-docs: docs/reference/pbm-commands.md

| `replsets`  | The list of replica sets included in the restore. Each replica set has the following fields: <br> - `name` - the replica set name <br> - `status` - the restore status on this replica set <br> - `error` - the error message for failed restore <br> - `last_transition_time` - the human-readable indication of the time when the restore process changed its status <br> - `nodes` - the list of nodes included in the restore. |
| `replsets.nodes`     | The list of nodes included in the restore. Each node has the following fields: <br> - `name` - the node name and port <br> - `status` - the restore status on the node <br> - `error` - the error message for failed restore <br> - `last_transition_time` - the human-readable indication of the time when the restore process changed its status |

## pbm diagnostic

> Generates the report with the detailed information about a specified logical backup or a restore. The report includes the following information:

Reviewer: ```suggestion
Generates the report with detailed information about command progress that is useful for reproducing reported issues. The report includes the following information:
```

Author: Applied the latter suggestion. Resolving

> * The information about the environment where a backup or a restore is running. 
> * Logs collected between the start and end time of the command execution
> * Metadata file of a specified backup or a restore

Reviewer: ```suggestion
* The information about the environment: PBM Agents statuses and cluster members. 
* Logs collected between the start and end time of the command execution
* If it is a backup command, the backup metadata file.
* If it is a restore command, the restore metadata file and the backup metadata file. (NOTE: Physical Restore is not supported at the moment)
```

The command has the following syntax:

```{.bash data-prompt="$"}
$ pbm diagnose --path path --name <backup-name> --opid <OPID>
```

The command accepts the following flags:

| Flag                | Description                      |
| ------------------- | -------------------------------- |
| `--path`            | The path where to save the report. If the directory doesn’t exist, PBM creates it during the report generation. Make sure that the user that runs PBM CLI has write access to the specified path |

> | `--name`            | The name of the required backup or a restore |
> | `--opid`            | The unique Operation ID of the specified backup or a restore. You can retrieve it from the `pbm describe-backup` / `pbm describe-restore` output. |
> | `--archive`         | Creates teh .zip archive of the report on the specified path.|

Reviewer: ```suggestion
| `--name`            | The name of the backup or restore (optional) |
| `--opid` | The unique Operation ID of the command. You can retrieve it from the `pbm describe-backup` / `pbm describe-restore` output or from `pbm logs` command. |
| `--archive` | Creates zip archive of the report in the path.|
```

Author: Updated

| `replsets`  | The list of replica sets included in the restore. Each replica set has the following fields: <br> - `name` - the replica set name <br> - `status` - the restore status on this replica set <br> - `error` - the error message for failed restore <br> - `last_transition_time` - the human-readable indication of the time when the restore process changed its status <br> - `nodes` - the list of nodes included in the restore. |
| `replsets.nodes`     | The list of nodes included in the restore. Each node has the following fields: <br> - `name` - the node name and port <br> - `status` - the restore status on the node <br> - `error` - the error message for failed restore <br> - `last_transition_time` - the human-readable indication of the time when the restore process changed its status |

## pbm diagnostic

> Generates the report with the detailed information about a a specific backup, restore, or other commands. The report includes the following information:

Reviewer 2: I'm against using "report" - as there's no report generated - it doesn't contain any answers or analysis, just a set of diagnostic files.  Let's call it a "diagnostic archive".

```suggestion
Generates a diagnostic archive with detailed information about a specific backup or restore. The archive includes the following information:
```

Reviewer: without --archive file, it will be just files.

the command also works for other commands. it is jot specific for backup or restore

Reviewer 2: maybe we should drop "--archive" and always create a zip archive ?

Reviewer 2: Eventually we may name it "Diagnostic Dump"

Author: So what's the outcome of this discussion?:)

Author: I'd stick to report. An archive is also a report, only archived

> * The information about the environment: pbm-agents statuses, cluster members, etc. 

Reviewer 2: I think we may skip most of the introduction here as it's described in the dedicated page. I think 1 sentence about what it does, may be enough. Then we can proceed to details with different parameters.

Author: Reworded, left the link to the feature description.

* Logs collected between the start and end time of the command execution
* If it is a backup command, the backup metadata file.
* If it is a restore command, the restore metadata file and the backup metadata file. 

   <i warning>:material-alert: Warning:</i> Physical restore is not supported in version 2.8.0.

> To learn more, refer to the [Diagnostics report](../troubleshoot/pbm-report.md).

Reviewer 2: ```suggestion
To learn more, refer to the [Diagnostics archive](../troubleshoot/pbm-report.md).
```

Reviewer: again, to generate a zip archive you need to explicitly ask by providing --archive flag

The command has the following syntax:

```{.bash data-prompt="$"}
$ pbm diagnose --path path --name <backup-name> --opid <OPID>
```

The command accepts the following flags:

| Flag                | Description                      |
| ------------------- | -------------------------------- |

> | `--path`            | The path where to save the report. If the directory doesn’t exist, PBM creates it during the report generation. Make sure that the user that runs PBM CLI has write access to the specified path |

Reviewer 2: ```suggestion
| `--path`            | The path where to save the diagnostic archive. If the directory doesn’t exist, PBM creates it during the archive generation. Make sure that the user that runs PBM CLI has write access to the specified path |
```

| `--name`            | The name of the required backup or a restore |
| `--opid`            | The unique Operation ID of the specified command. You can retrieve it from the `pbm logs`, `pbm describe-backup` / `pbm describe-restore` output. |

> | `--archive`         | Creates teh .zip archive of the report in the specified path.|

Reviewer 2: ```suggestion
| `--archive`         | Creates a .zip archive of the report in the specified path.|
```
